### PR TITLE
Fix pedantic errors

### PR DESCRIPTION
**Pull request by stschiff**

### CODE REVIEW

**Comment by stschiff** [comment]
Uff, this was quite a bit of stuff.

@nevrome please pay particular attention to some of the shadowing that I've resolved (I've added trailing underscores where a variable shadows one with the same name). There was one place with `loadedKB` or `loadedB` or so where I think there was a bug.

File: src/Poseidon/CLI/Fetch.hs [hunk 1]
```diff
@@ -187,19 +188,19 @@ printDownloadProgress fileSizeMB = loop 0 0
             x <- await
             maybe (return ()) (showDownloaded fileSizeMB loadedB) x
             where
-                showDownloaded fileSizeMB loadedKB x = do
-                    let newLoadedB = loadedB + B.length x
+                showDownloaded fileSizeMB_ loadedB_ x = do
```
Comment (stschiff) [inline src/Poseidon/CLI/Fetch.hs:191]: Here and in the following: Please check @nevrome whether this is all correct.

Comment (nevrome) [reply src/Poseidon/CLI/Fetch.hs:191]: Good spotting! Not sure why that worked, but your solution works as well, so I will just accept that :shrug:.

**Comment by nevrome** [comment]
Looks all fine to me - I also ran some tests and all still worked as expected

**Comment by stschiff** [comment]
OK, I'll merge.

**Pull request merged**